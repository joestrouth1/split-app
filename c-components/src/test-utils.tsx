/**@jsx jsx */
import { render, RenderOptions } from '@testing-library/react'
import { jsx, ThemeProvider } from 'theme-ui'
import { ReactNode, ReactElement } from 'react'
import { defaultTheme } from './theme'

const Wrapped = (props: { children?: ReactNode }) => {
  return <ThemeProvider theme={defaultTheme} {...props} />
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapped, ...options })

export * from '@testing-library/react'

export { customRender as render }
