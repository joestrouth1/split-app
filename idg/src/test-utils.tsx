/**@jsx jsx */
import { render, RenderOptions } from '@testing-library/react'
import { jsx, ThemeProvider } from 'theme-ui'
import { ReactNode, ReactElement } from 'react'
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from '@reach/router'
import { defaultTheme } from 'c-components'

const Wrapped = (props: { children?: ReactNode }) => {
  return <ThemeProvider theme={defaultTheme} {...props} />
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapped, ...options })

function renderWithRouter(
  ui: ReactElement,
  { route = '/', history = createHistory(createMemorySource(route)) }
) {
  return {
    ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
    history,
  }
}

export * from '@testing-library/react'

export { customRender as render, renderWithRouter }
