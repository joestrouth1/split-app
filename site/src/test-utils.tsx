/**@jsx jsx */
import { render, RenderOptions } from '@testing-library/react'
import { jsx, ThemeProvider } from 'theme-ui'
import { ReactNode, ReactElement } from 'react'
import { HelmetProvider } from 'react-helmet-async'
import { RoutingContext, useRoutingState } from './contexts/routing'
import { defaultTheme } from 'components'

// eslint-disable-next-line
//@ts-ignore
HelmetProvider.canUseDOM = false

const Wrapped = ({ children }: { children?: ReactNode }) => {
  const helmetContext = {}
  const routingState = useRoutingState()
  return (
    <ThemeProvider theme={defaultTheme}>
      <HelmetProvider context={helmetContext}>
        <RoutingContext.Provider value={routingState}>
          {children}
        </RoutingContext.Provider>
      </HelmetProvider>
    </ThemeProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapped, ...options })

export * from '@testing-library/react'

export { customRender as render }
