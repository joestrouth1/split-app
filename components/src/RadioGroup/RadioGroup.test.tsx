/**@jsx jsx */
import { jsx } from 'theme-ui'
import { render } from '../test-utils'
import { RadioGroup } from '.'
// import { RadioOption } from '../RadioOption'

describe('RadioGroup', () => {
  it('renders without throwing', () => {
    expect(() =>
      render(<RadioGroup name="test">Text</RadioGroup>)
    ).not.toThrow()
  })
})
