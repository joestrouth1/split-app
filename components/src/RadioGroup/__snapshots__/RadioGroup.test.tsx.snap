// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders without throwing 1`] = `
<DocumentFragment>
  <div
    class="css-1qol7jq"
  >
    <div
      class="css-80zs6q"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;
